src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,883 @@ const ActorInit Object_Kankyo_InitVars = {
     (ActorFunc)ObjectKankyo_Update,
     (ActorFunc)ObjectKankyo_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Object_Kankyo/ObjectKankyo_Init.s")
+static u8 sIsSpawned = false;
+static s16 sTrailingFairies = 0;
+
+void ObjectKankyo_SetupAction(ObjectKankyo* this, ObjectKankyoActionFunc action) {
+    this->actionFunc = action;
+}
+
+void ObjectKankyo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjectKankyo* this = THIS;
+    s32 pad;
+    s16 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        this->effects[i].state = 0;
+    }
+
+    this->actor.room = -1;
+    switch (this->actor.params) {
+        case 0:
+            if (!sIsSpawned) {
```

Comment:
This is a pattern used in a few actors so the name is fine imo. As you mention, the issue is that some variants aren't using it when they should. Which will need to be an @bug

Reply:
Right. I forgot this actor is good at making obstacles not spawn 😎

I tried to investigate variable 2 "lightning" see if it could also be duped but it's only used in an alternate header for a cutscene apparently so maybe not